dir.rb

```diff
@@ -1,11 +1,85 @@
-# Objects of class Dir are directory streams representing
-# directories in the underlying file system. They provide a variety
-# of ways to list directories and their contents. See also File.
+# An object of class \Dir represents a directory in the underlying file system.
 #
-# The directory used in these examples contains the two regular files
-# (<code>config.h</code> and <code>main.rb</code>), the parent
-# directory (<code>..</code>), and the directory itself
-# (<code>.</code>).
+# It consists mainly of:
+#
+# - A string _path_, given when the object is created,
+#   that specifies a directory in the underlying file system;
+#   method #path returns the path.
+# - A collection of string <i>entry names</i>,
+#   each of which is the name of a directory or file in the underlying file system;
+#   the entry names may be retrieved
+#   in an {array-like fashion}[rdoc-ref:Dir@Dir+As+Array-Like]
+#   or in a {stream-like fashion}[rdoc-ref:Dir@Dir+As+Stream-Like].
+#
+# == About the Examples
+#
+# Some examples on this page use this simple file tree:
+#
+#   example/
+#   ├── config.h
+#   ├── lib/
+#   │   ├── song/
+#   │   │   └── karaoke.rb
+#   │   └── song.rb
+#   └── main.rb
+#
+# Others use the file tree for the
+# {Ruby project itself}[https://github.com/ruby/ruby].
+#
+# == \Dir As \Array-Like
+#
+# A \Dir object is in some ways array-like:
+#
+# - It has instance methods #children, #each, and #each_child.
+# - It includes {module Enumerable}[rdoc-ref:Enumerable@What-27s+Here].
+#
+# == \Dir As Stream-Like
+#
+# A \Dir object is in some ways stream-like.
+#
+# The stream is initially open for reading,
+# but may be closed manually (using method #close),
+# and will be closed on block exit if created by Dir.open called with a block.
+# The closed stream may not be further manipulated,
+# and may not be reopened.
+#
+# The stream has a _position_, which is the index of an entry in the directory:
+#
+# - The initial position is zero (before the first entry).
+# - \Method #tell (aliased as #pos) returns the position.
+# - \Method #pos= sets the position (but ignores a value outside the stream),
+#   and returns the position.
+# - \Method #seek is like #pos, but returns +self+ (convenient for chaining).
```

Comment:
I think `#seek` is more like `#pos=` right? It sets the position rather than gets the position.

Reply:
Fixed.